File: 프로그래머스/unrated/181830. 정사각형으로 만들기/정사각형으로 만들기.js
function solution(arr) {
    if (arr.length === arr[0].length) {
        return arr;
    } else if (arr.length > arr[0].length) {
        const count = arr.length - arr[0].length;
        
        arr.forEach(el => {
            for (let i = 0; i < count; i++) {
                el.push(0);
            }
        });
        
        return arr;
    } else {
        const count = arr[0].length - arr.length;
        
        for (let i = 0; i < count; i++) {
            arr.push(new Array(arr[0].length).fill(0));
        }
        
        return arr;
    }
}

File: 프로그래머스/unrated/181830. 정사각형으로 만들기/정사각형으로 만들기.java
class Solution {
    public int[][] solution(int[][] arr) {
        int rowCount = arr.length;
        int colCount = arr[0].length;
        
        if (rowCount == colCount) {
            return arr;
        }
        
        int[][] result;
        if (rowCount > colCount) {
            result = new int[rowCount][rowCount];
        } else {
            result = new int[colCount][colCount];
        }
        
        for (int i = 0; i < rowCount; i++) {
            for (int j = 0; j < colCount; j++) {
                result[i][j] = arr[i][j];
            }
        }
        return result;
    }
}